// Copyright 2022 TIER IV, INC. All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

service OtaClientIoTLoggingService {
  /*
   * `PutLog` service requests OTA Client logging service to put log.
   */
  rpc PutLog(PutLogRequest) returns (PutLogResponse) {}
}

enum LogType {
  LOG = 0;
  METRICS = 1;
}

enum ErrorCode {
  NO_FAILURE = 0;         // Success
  SERVER_QUEUE_FULL = 1;  // Error: Server queue is full
  NOT_ALLOWED_ECU_ID = 2; // Error: Specified ECU ID is not allowed
  NO_MESSAGE = 3;         // Error: No message in the request
}

message PutLogRequest {
  string ecu_id = 1;    // target ECU ID
  LogType log_type = 2; // log type
  string message = 3;   // log message
}

message PutLogResponse {
  ErrorCode code = 1; // error code
  string message = 2; // error message
}
